blockchain/contracts/MainContract.sol

```diff
@@ -128,12 +123,15 @@ contract NFTtheWorld {
         } else {
             maxNumberOfNFTsToBuy[dropHash] = (_uris.length * 5) / 100;
         }
-        availableNFTsCount[dropHash] = availableNFTs[dropHash].length;
         numberOfDrops += 1;
     }
 
     // This function lets a user join a drop by specifying the number of NFTs she would like to reserve.
     function joinDrop(uint256 _numberOfNFTs, uint256 _dropHash) public {
+        require(
+            nftOwnerships[_dropHash][0].dropTime > block.timestamp,
```

Comment:
just a small thing: would be nicer if you would use dropData[_dropHash].dropTime here

Reply:
True, done